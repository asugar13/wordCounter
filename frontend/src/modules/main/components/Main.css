.selector {
  top: 70px;  
  width: 350px;
  margin-left: auto;
  margin-right: auto;
  position: relative;
}
.search{
  bottom: 62px;
  height: 50px;
  left: 355px;
  position: absolute
}
.title {
  display: flex;
  justify-content: center;
  margin-top: 50px;  
}

.urlHistoryList {
  left: 10%;
  width: 350px;
  top: 60px;  
  position: relative;
}

.table {
  margin-top: 120px;
  margin-left: auto;
  margin-right: auto;
  width: 80% !important;
}

.validation{
  bottom: 35px;
  position: absolute;}