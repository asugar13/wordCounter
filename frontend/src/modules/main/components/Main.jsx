import React, { useEffect, useState } from 'react';
import InputUrl from "./InputUrl"
import { Spinner } from "react-bootstrap";
import "./Main.css"
import NotificationManager from "react-notifications/lib/NotificationManager"
import { Grid, GridColumn } from '@progress/kendo-react-grid';
import { process } from '@progress/kendo-data-query'
import _ from "underscore"
import { CustomColumnMenu } from "./CustomColumnMenu"

export default ({
    wordsLoading,
    getWords,
    urlHistory,
    wordCounts
}) => {
    const [selectedUrl, setSelectedUrl] = useState(null)
    const [wordCountMappings, setWordCountMappings] = useState(null)
    // const [wordsLoading, setWordsLoading] = useState(false)
    // const [urlHistory, setUrlHistory] = useState([])
    // const [wordCounts, setWordCounts] = useState([])


    const columns = [{
        title: "Word",
        field: "word",
        show: true
    },
    {
        title: "Count",
        field: "count",
        show: true
    }]

    let arg = {
        skip: 0,
        take: 25,
        sort: [
            { field: 'count', dir: 'desc' }
        ],
    }
    const [dataState, setDataState] = useState(arg)
    const [dataResult, setDataResult] = useState(null)

    // const getWords = () => {
    //     const url = "http://localhost:3000/main/word-count"
    //     let data = {
    //         request: selectedUrl,
    //     };
    //     const otherParam = {
    //         headers: { 'Content-Type': 'application/json' },
    //         body: JSON.stringify(data),
    //         method: "POST"
    //     };
    //     let status
    //     setWordsLoading(true)
    //     fetch(url, otherParam)
    //         .then(response => {
    //             status = response.status
    //             setWordsLoading(false)
    //             return response.json()
    //         })
    //         .then(ans => {
    //             if (status !== 200) {
    //                 setWordCounts([])
    //                 return NotificationManager.error("No Response")
    //             }
    //             let requestOption = { value: selectedUrl, label: selectedUrl }
    //             let labels = _.pluck(urlHistory, "label")
    //             if (labels.indexOf(selectedUrl) !== -1) {
    //                 let newHistory = urlHistory
    //                 newHistory.splice(labels.indexOf(selectedUrl), 1)
    //                 setUrlHistory([...newHistory, requestOption])
    //             } else {
    //                 setUrlHistory(history => [...history, requestOption])
    //             }
    //             return setWordCounts(ans)
    //         })
    //         .catch(e => {
    //             setWordsLoading(false)
    //             console.log(e)
    //         });
    // }

    useEffect(() => {
        if (selectedUrl) {
            getWords(selectedUrl)
            // getWords()
        }
    }, [selectedUrl])

    useEffect(() => {
        let formattedWordCounts = Object.keys(wordCounts).map(word => {
            return {
                count: wordCounts[word],
                word: word
            }
        })
        setWordCountMappings(formattedWordCounts)
        setDataResult(process(formattedWordCounts, dataState))

    }, [wordCounts])

    const dataStateChange = (event) => {
        setDataResult(process(wordCountMappings, event.dataState))
        setDataState(event.dataState)
    }

    return <>
        <h3 className="title">Enter a URL to find out how many times each word appears in its content</h3>
        <InputUrl urlHistory={urlHistory} setSelectedUrl={setSelectedUrl} />

        <div className="table">
            {
                wordsLoading ?
                    <Spinner animation="border" role="status">
                        <span className="sr-only">Loading...</span>
                    </Spinner> :
                    <Grid
                        style={{ height: '700px' }}
                        sortable
                        pageable={{ buttonCount: 4, pageSizes: true }}
                        data={dataResult}
                        {...dataState}
                        onDataStateChange={(event) => dataStateChange(event)}
                    >
                        {_.map(columns, (column, index) => {

                            return (
                                <GridColumn
                                    key={index}
                                    field={column.field}
                                    title={column.title}
                                    columnMenu={
                                        props =>
                                            <CustomColumnMenu
                                                {...props}
                                                columns={columns}
                                            />
                                    }
                                />
                            )
                        })}
                    </Grid>
            }
        </div>
    </>
}
