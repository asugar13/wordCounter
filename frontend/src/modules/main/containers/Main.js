import Main from "../components/Main"
import {
    connect
} from "react-redux"
import {
    getWords
} from "../actions/main"

const mapStateToProps = (state, ownProps) => {
    return {
        wordsLoading: state.wordsLoading,
        urlHistory: state.urlHistory,
        wordCounts: state.wordCounts
    }
}

const mapDispatchToProps = dispatch => {
    return {
        getWords: (url) => dispatch(getWords(url))
    }
}

const MainContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(Main)

export default MainContainer