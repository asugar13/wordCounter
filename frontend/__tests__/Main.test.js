import React from 'react';
import Main from '../src/modules/main/components/Main';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

//Uncomment the css import in Main.jsx to test this component. 
describe("<Main /> component", () => {
  const mainContainer = shallow(<Main />);
  // it('should match the snapshot', () => {
  //   expect(mainContainer.html()).toMatchSnapshot()
  // })

  it("accepts wordsLoading and wordCounts props", () => {
    const wrapper = mount(<Main wordsLoading={false} wordCounts={{ "a": 1 }} />);
    expect(wrapper.props().wordsLoading).toEqual(false);
    expect(wrapper.props().wordCounts).toEqual({ "a": 1 });

  });

  it('should have a div with className table', () => {
    expect(mainContainer.find('[className="table"]').length).toEqual(1)
  });

  it('should have a div with className title', () => {
    expect(mainContainer.find('[className="title"]').length).toEqual(1)
  });
})