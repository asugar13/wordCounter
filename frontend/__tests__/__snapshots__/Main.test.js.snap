// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> with empty urlHistory as prop should match the snapshot 1`] = `"<h3 class=\\"title\\">Enter a URL to find out how many times each word appears in its content</h3><div class=\\"selector\\"><form><div class=\\"MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon\\" role=\\"combobox\\" aria-expanded=\\"false\\" placeholder=\\"Enter a valid URL\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" id=\\"undefined-label\\">Enter a valid URL</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd\\"><input type=\\"text\\" aria-invalid=\\"false\\" autoComplete=\\"off\\" name=\\"url\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd\\" aria-autocomplete=\\"list\\" autoCapitalize=\\"none\\" spellcheck=\\"false\\"/><div class=\\"MuiAutocomplete-endAdornment\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Clear\\" title=\\"Clear\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Open\\" title=\\"Open\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></span></button></div><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Enter a valid URL</span></legend></fieldset></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained search\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Search</span></button></form></div><div class=\\"table\\"><div style=\\"height:700px\\" class=\\"k-widget k-grid\\" role=\\"grid\\" aria-colcount=\\"2\\" aria-rowcount=\\"0\\" data-keyboardnavscope=\\"true\\"><div class=\\"k-grid-header\\" role=\\"presentation\\"><div class=\\"k-grid-header-wrap\\" role=\\"presentation\\" style=\\"border-width:0\\"><table role=\\"presentation\\"><colgroup role=\\"presentation\\"><col/><col/></colgroup><thead role=\\"presentation\\" data-keyboardnavheader=\\"true\\"><tr style=\\"touch-action:none\\" role=\\"row\\" aria-rowindex=\\"1\\"><th aria-sort=\\"none\\" aria-colindex=\\"1\\" aria-selected=\\"false\\" colSpan=\\"1\\" rowspan=\\"1\\" class=\\"k-header \\" role=\\"columnheader\\"><div style=\\"display:inline\\"><div style=\\"display:inline;position:absolute;cursor:pointer;right:5px\\"><span class=\\"k-icon k-i-more-vertical\\"></span></div></div><span class=\\"k-link\\">Word<span aria-label=\\"Sortable\\"></span></span></th><th aria-sort=\\"descending\\" aria-colindex=\\"2\\" aria-selected=\\"false\\" colSpan=\\"1\\" rowspan=\\"1\\" class=\\"k-header  k-sorted\\" role=\\"columnheader\\"><div style=\\"display:inline\\"><div style=\\"display:inline;position:absolute;cursor:pointer;right:5px\\"><span class=\\"k-icon k-i-more-vertical\\"></span></div></div><span class=\\"k-link\\">Count<span aria-label=\\"Sortable\\" class=\\"k-icon k-i-sort-desc-sm\\"></span></span></th></tr></thead></table></div></div><div class=\\"k-grid-container\\" role=\\"presentation\\"><div class=\\"k-grid-content k-virtual-content\\" role=\\"presentation\\"><div style=\\"position:relative\\" role=\\"presentation\\"><table class=\\"k-grid-table\\" role=\\"presentation\\"><colgroup role=\\"presentation\\"><col/><col class=\\"k-sorted\\"/></colgroup><tbody role=\\"presentation\\" data-keyboardnavbody=\\"true\\"><tr class=\\"k-grid-norecords\\" aria-rowindex=\\"2\\"><td colSpan=\\"2\\">No records available</td></tr></tbody></table></div><div class=\\"k-height-container\\" role=\\"presentation\\"><div></div></div></div></div><div class=\\"k-pager-wrap k-pager k-widget k-grid-pager\\" role=\\"navigation\\"><a href=\\"#\\" class=\\"k-link k-pager-nav k-pager-first k-state-disabled\\" title=\\"Go to the first page\\"><span class=\\"k-icon k-i-arrow-end-left\\" aria-label=\\"Go to the first page\\"></span></a><a href=\\"#\\" class=\\"k-link k-pager-nav k-state-disabled\\" title=\\"Go to the previous page\\"><span class=\\"k-icon k-i-arrow-60-left\\" aria-label=\\"Go to the previous page\\"></span></a><div class=\\"k-pager-numbers-wrap\\"><select class=\\"k-dropdown\\"></select><ul class=\\"k-pager-numbers k-reset\\"></ul></div><a href=\\"#\\" class=\\"k-link k-pager-nav k-state-disabled\\" title=\\"Go to the next page\\"><span class=\\"k-icon k-i-arrow-60-right\\" aria-label=\\"Go to the next page\\"></span></a><a href=\\"#\\" class=\\"k-link k-pager-nav k-pager-last k-state-disabled\\" title=\\"Go to the last page\\"><span class=\\"k-icon k-i-arrow-end-right\\" aria-label=\\"Go to the last page\\"></span></a><span class=\\"k-pager-sizes k-label\\"><span class=\\"k-widget k-dropdown\\"><span role=\\"listbox\\" tabindex=\\"0\\" class=\\"k-dropdown-wrap\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" aria-owns=\\"a56a6359-26d2-4d3c-841e-e56846e26360\\"><span class=\\"k-input\\">25</span><span class=\\"k-select\\"><span class=\\"k-icon k-i-arrow-s\\"></span></span><select tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"opacity:0;width:1px;border:0;z-index:-1;position:absolute;left:50%\\"><option value=\\"25\\"></option></select></span></span>items per page</span><div class=\\"k-pager-info k-label\\">0 - 0 of 0 items</div></div></div></div>"`;
